lib/model/rwfolder.go

```diff
@@ -262,11 +264,12 @@ func (f *sendReceiveFolder) pull(prevSeq int64, prevIgnoreHash string) (curSeq i
 	f.clearErrors()
 	var changed int
 	tries := 0
+	toBeScanned := make(map[string]struct{})
```

Comment:
This map appears to be passed around everywhere during the pulling. Given we have multiple pullers this should be a data race, probably with a crash when the runtime detects simultaneous modification. I think we just don't tickle this in the tests...

I suggest a similar mechanism to the db updater thing. A channel for items to be scanned (that can be passed around willy nilly), a loop that collects items from it, channel is closed when we're done, the loop exits and does the scan.